### DISCUSSION TITLE
Adding a token

### DISCUSSION THREAD

**Discussion by michalip** [jenkinsci/bitbucket-push-and-pull-request-plugin#209] [Ideas]
Hi 
Any chance of adding a token which would assist targeting webhook->specific job ?

thanks

**Comment by macghriogair** [comment]
Hey @michalip, could you elaborate a bit on this? What is the use case? Triggering different jobs by having different webhook urls (e.g. with GET-Parameter) set in bitbucket? afaik it is not possible to send custom request headers with the bitbubcket hook, is it?

**Comment by michalip** [comment]
Hi @macghriogair , 

Use case: I've used BPPP to handle master branch only. Whatever happens on master branch, BPPP received the webhook , builds , presents the status on Build monitor.
If I want to have a SINGLE catch all job for FEATURES, I need a second webhook in all repos with a TOKEN/Secret -> pointing at that single catch all job.

AFAIK , this is possible with secret/tokens

I'm not sure I can provide more info, besides the use case , which is done so that work is reduced and the daily picture (master) is preserved and presented to the developers without changing for each minor change  in feature branches

note: 

1. There is a field in bitbucket to enter this token/secret
2. There are other plugins for bitbucket, gitlab etc that support this functionality. Some use it to help with the resolution of the job, others for security(?). If needed I can provide links

**Comment by macghriogair** [comment]
Thanks for the clarification. Could u provide a link to some sample plugin implementing this?

**Comment by macghriogair** [comment]
Another question: could you not use the branch filter (allowed branches: `feature/*`) to achieve what you describe or am I missing sth.?

**Comment by michalip** [comment]
Hi @macghriogair , 

The sample plugins implementing tokens:

1. [Generic](https://github.com/jenkinsci/generic-webhook-trigger-plugin)
2. [Gitlab ](https://github.com/jenkinsci/gitlab-plugin)

To reiterate, the token provides more security and also linking specific repos to specific jobs in Jenkins.

I'm not a Jenkins expert; from what I understand when an incoming payload hits Jenkins, Jenkins tries to match the incoming repo with a job that has build that repo in the past. So creating only one job to handle many incoming requests without a token would not work. 

If it's something you can implement, I m sure myself and others would appreciate.

Hope that helps and thanks for checking

**Comment by macghriogair** [comment]
the security point alone should be sufficient to have this as a feature. 👍
